### DISCUSSION TITLE
How to use a .hexpat file?

### DISCUSSION THREAD

**Discussion by alexspurling** [WerWolv/ImHex#1478] [Get Help] [answered]
I would like to load a .bmp file and view it using the BMP .hexpat pattern file available here: https://raw.githubusercontent.com/WerWolv/ImHex-Patterns/master/patterns/bmp.hexpat

I cannot find any option in ImHex to load this .hexpat file. If I try to copy and paste the contents of the file into the Pattern Editor view, nothing happens (no text is pasted). How do I load the pattern file?

**Comment by WerWolv** [comment, marked as answer]
Hey! To import a pattern, you can go to File -> Import -> Pattern and then select it from the list there. That will automatically execute it.
If you're copy-pasting the content into the pattern editor (or write your own pattern in there later on) you need to press the Play Button at the bottom of the editor to execute the pattern. 
<img width="76" alt="Screenshot 2023-12-30 at 12 24 33" src="https://github.com/WerWolv/ImHex/assets/10835354/c05aa896-e953-49f7-8a7f-1cca9fdd9a33">
